File: webapp/Locus.Core/Services/RequirementsTestService.cs
using Locus.Core.Models;
using Locus.Core.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Locus.Core.Services
{
    public interface IRequirementsTestService
    {

        RequirementsTest Save(RequirementsTest rt);

        RequirementsTest Update(RequirementsTest rt);

        RequirementsTest Delete(int idrt);

        List<RequirementsTest> GetAll();

        RequirementsTest Get(int idrt);

        List<RequirementsTest> GetTestCaseRelations(int id);

        RequirementsTest DeleteTestCase(int reqId, int tcId);

        RequirementsTest DeleteTestProcedure(int reqId, int tpId);

        RequirementsTest DeleteTestScenario(int reqId, int tsId);
        List<RequirementsTest> GetTestProcedureRelations(int id);
        List<RequirementsTest> GetTestScenarioRelations(int id);
    }
    public class RequirementsTestService : IRequirementsTestService
    {
        private readonly IRequirementsTestRepository _rtRepository;

        public RequirementsTestService(IRequirementsTestRepository rtRepository)
        {
            _rtRepository = rtRepository;
        }

  
        public RequirementsTest Delete(int idrt)
        {
           return _rtRepository.Delete(idrt);
        }

        public RequirementsTest DeleteTestCase(int reqId, int tcId)
        {
            return _rtRepository.DeleteTestCase(reqId, tcId);
        }

        public RequirementsTest DeleteTestProcedure(int reqId, int tpId)
        {
            return _rtRepository.DeleteTestProcedure(reqId, tpId);
        }

        public RequirementsTest DeleteTestScenario(int reqId, int tsId)
        {
            return _rtRepository.DeleteTestScenario(reqId, tsId);
        }

        public RequirementsTest Get(int idrt)
        {
            return _rtRepository.Get(idrt);
        }

        public List<RequirementsTest> GetAll()
        {
            return _rtRepository.GetAll();
        }

        public List<RequirementsTest> GetTestCaseRelations(int id)
        {
            return _rtRepository.GetTestCaseRelations(id);
        }

        public List<RequirementsTest> GetTestProcedureRelations(int id)
        {
            return _rtRepository.GetTestProcedureRelations(id);
        }

        public List<RequirementsTest> GetTestScenarioRelations(int id)
        {
            return _rtRepository.GetTestScenarioRelations(id);
        }

        public RequirementsTest Save(RequirementsTest rt)
        {
            return _rtRepository.Save(rt);
        }

        public RequirementsTest Update(RequirementsTest rt)
        {
            return _rtRepository.Update(rt);
        }
    }
}

File: webapp/Locust/Scripts/app/services/RequirementsTestService.js
(function (angular, undefined) {
    'use strict';

    // Get services module
    var app = angular.module(appName);

    app.service('RequirementsTestService', ['$resource', function ($resource) {
        
        var noop = angular.noop;
        var resource = $resource('/api/RequirementsTest/:action/:id',
            { id: '@Id' }, {
                Save: {
                    method: 'POST'
                },
                Update: {
                    method: 'PUT'
                },
                Delete: {
                    method: 'DELETE',
                    params: {
                        id: "@id"
                    }
                },
                Get: {
                    method: 'GET',
                    isArray: false
                },
                GetAll: {
                    method: 'GET',
                    isArray: true
                },
                GetTestCaseRelations: {
                    method: 'GET',
                    isArray: true
                },
                GetTestScenarioRelations: {
                    method: 'GET',
                    isArray: true
                },
                GetTestProcedureRelations: {
                    method: 'GET',
                    isArray: true
                },
                DeleteTestCase: {
                    method: 'GET',
                    isArray: false
                },
                DeleteTestProcedure: {
                    method: 'GET',
                    isArray: false
                },
                DeleteTestScenario: {
                    method: 'GET',
                    isArray: false
                }
            }),

            self = this;

        self.GetAll = function (success, error) {
            return resource.GetAll({ action: 'GetAll' }, success || noop, error || noop);
        };

        self.Get = function (id, success, error) {
            return resource.Get({ action: 'Get', id: id }, success || noop, error || noop);
        };

        self.Delete = function (id, success, error) {
            return resource.Delete({ action: 'Delete', id: id }, success || noop, error || noop);
        };

        self.DeleteTestCase = function (reqId, tcId, success, error) {
            return resource.DeleteTestCase({ action: 'DeleteTestCase', reqId: reqId, tcId: tcId }, success || noop, error || noop);
        };

        self.DeleteTestProcedure = function (reqId, tpId, success, error) {
            return resource.DeleteTestProcedure({ action: 'DeleteTestProcedure', reqId: reqId, tpId: tpId }, success || noop, error || noop);
        };

        self.DeleteTestScenario = function (reqId, tsId, success, error) {
            return resource.DeleteTestScenario({ action: 'DeleteTestScenario', reqId: reqId, tsId: tsId }, success || noop, error || noop);
        };

        self.GetTestCaseRelations = function (id, success, error) {
            return resource.GetTestCaseRelations({ action: 'GetTestCaseRelations', id: id }, success || noop, error || noop)
        };
        self.GetTestProcedureRelations = function (id, success, error) {
            return resource.GetTestProcedureRelations({ action: 'GetTestProcedureRelations', id: id }, success || noop, error || noop)
        };
        self.GetTestScenarioRelations = function (id, success, error) {
            return resource.GetTestScenarioRelations({ action: 'GetTestScenarioRelations', id: id }, success || noop, error || noop)
        };
        self.Save = function (entity, success, error) {
            return resource.Save({ action: 'Save' }, entity, success || noop, error || noop);
        };

        self.Update = function (entity, success, error) {
            return resource.Update({ action: 'Update' }, entity, success || noop, error || noop);
        };
        
    }]);
})(angular);